src/components.md

```diff
@@ -72,23 +72,11 @@ The message type allows us to switch between the modes.
 {{#include ../examples/components.rs:header_msg }}
 ```
 
-For components we also need to implement the `Model` trait. The `Components` type is empty here because it refers to child components. We don't have any child components for our header bar so we use a `()`.
-
-```rust,no_run,noplayground
-{{#include ../examples/components.rs:header_model_impl }}
-```
-
-The update function is rather minimal. If our header bar was more complex, storing state in this component would make sense, but because we just handle a few buttons, we can simply forward messages. For that we can use the `parent_sender`. You can see that the message type of the main application is `AppMsg` and that there's an enum `AppMode`. Both were not introduced yet, but will be explained later. For now, we just need to know that this component will send `SetMode` messages to the app.
-
-```rust,no_run,noplayground
-{{#include ../examples/components.rs:header_update }}
-```
-
-> We don't use the `_parent_model` argument of the `init_model` in this example. Yet you can use it if you need to access information from the parent model during initialization, for example for passing a resource shared with the component.
+Our component needs no `update` method, because the `view` can emit the component's output messages as part of its click signal handlers, as we will see in the next section.
 
 ### The widgets
 
-There's nothing special about widgets of a component. The only difference to the main app is that the root widget doesn't need to be a `gtk::ApplicationWindow`. Instead, we use a `gtk::HeaderBar` here, but theoretically the root widget doesn't even need to be a widget at all (which can be useful in special cases).
+There's nothing special about widgets of a child component. The only difference to the main app component is that the root widget doesn't need to be a `gtk::ApplicationWindow`. Instead, we use a `gtk::HeaderBar` here, but theoretically the root widget doesn't even need to be a widget at all (which can be useful in special cases).
```

Comment:
Another thing that should be changed here: with 0.5 you use `gtk::Window` instead of `gtk::ApplicationWindow` as root for the main app component.